Source/WebKit/WebProcess/Cache/WebCacheStorageConnection.cpp

```diff
@@ -67,24 +67,32 @@ auto WebCacheStorageConnection::remove(WebCore::DOMCacheIdentifier cacheIdentifi
     });
 }
 
-void WebCacheStorageConnection::retrieveCaches(const WebCore::ClientOrigin& origin, uint64_t updateCounter, WebCore::DOMCacheEngine::CacheInfosCallback&& callback)
+auto WebCacheStorageConnection::retrieveCaches(const WebCore::ClientOrigin& origin, uint64_t updateCounter) -> Ref<RetrieveCachesPromise>
 {
-    connection().sendWithAsyncReply(Messages::NetworkStorageManager::CacheStorageAllCaches(origin, updateCounter), WTFMove(callback));
+    return connection().sendWithPromisedReply(Messages::NetworkStorageManager::CacheStorageAllCaches(origin, updateCounter))->whenSettled(RunLoop::current(), [](auto&& result) {
+        return result ? RetrieveCachesPromise::createAndSettle(WTFMove(result.value())) : RetrieveCachesPromise::createAndReject(WebCore::DOMCacheEngine::Error::Internal);
```

Comment:
Why are we rejecting with `DOMCacheEngine::Error::Internal` instead of `result.error()`?

Reply:
> Why are we rejecting with `DOMCacheEngine::Error::Internal` instead of `result.error()`?

sendWithPromiseReply returns an Expected<Expected<T,F>, IPC::ErrorCode>